heat/core/dndarray.py

```diff
@@ -2473,50 +2493,84 @@ def resplit_(self, axis=None):
+                del new_arr
+                return self
+            return new_arr
         # tensor needs be split/sliced locally
-        elif self.split is None:
+        if self.split is None:
+            new_arr = self.copy() if not in_place else self
             _, _, slices = self.comm.chunk(self.shape, axis)
-            temp = self.__array[slices]
-            self.__array = torch.empty((1,), device=self.device.torch_device)
+            temp = new_arr.__array[slices]
+            new_arr.__array = torch.empty((1,), device=self.device.torch_device)
             # necessary to clear storage of local __array
-            self.__array = temp.clone().detach()
-            self.__split = axis
-
-        # entirely new split axis, need to redistribute
-        else:
-            _, output_shape, _ = self.comm.chunk(self.shape, axis)
-            redistributed = torch.empty(
-                output_shape, dtype=self.dtype.torch_type(), device=self.device.torch_device
-            )
-
-            send_counts, send_displs, _ = self.comm.counts_displs_shape(self.lshape, axis)
-            recv_counts, recv_displs, _ = self.comm.counts_displs_shape(self.shape, self.split)
-            self.comm.Alltoallv(
-                (self.__array, send_counts, send_displs),
-                (redistributed, recv_counts, recv_displs),
-                send_axis=axis,
-                recv_axis=self.split,
-            )
-
-            self.__array = redistributed
-            self.__split = axis
-
-        return self
+            new_arr.__array = temp.clone().detach()
+            new_arr.__split = axis
+            if in_place:
+                self.__array = new_arr._DNDarray__array
+                self.__gshape = new_arr.__gshape
+                self.__dtype = new_arr.__dtype
+                self.__split = new_arr.__split
+                self.__device = new_arr.__device
+                self.__comm = new_arr.__comm
+                del new_arr
+            return new_arr if not in_place else self
+
+        self.create_split_tiles()
+        new_arr = factories.empty(self.gshape, split=axis, dtype=self.dtype, device=self.device)
+        new_arr.create_split_tiles()
+        rank = self.comm.rank
+        waits = []
+        rcv_waits = {}
+        for rpr in range(self.comm.size):
+            # need to get where the tiles are on the new one first
+            # rpr is the destination
+            new_locs = torch.where(new_arr.tiles.tile_locations == rpr)
+            new_locs = torch.stack([new_locs[i] for i in range(self.numdims)], dim=1)
+
+            for i in range(new_locs.shape[0]):
+                key = tuple(new_locs[i].tolist())
+                spr = self.tiles.tile_locations[key].item()
+                to_send = self.tiles[key]
+                if spr == rank and spr != rpr:
+                    waits.append(self.comm.Isend(to_send.clone(), dest=rpr, tag=rank))
+                elif spr == rpr == rank:
```

Comment:
I am not sure that this is what you want here. This can only evaluate to True on rank 1 and above